app/modules/social_groups/parameters.py

```diff
@@ -35,11 +34,26 @@ class PatchSocialGroupDetailsParameters(PatchJSONParameters):
         PatchJSONParameters.OP_REMOVE,
     )
 
-    PATH_CHOICES = ('/name', '/members', '/member')
+    PATH_CHOICES = ('/name', '/members')
 
     @classmethod
     def add(cls, obj, field, value, state):
-        # For all fields, Add and replace are the same operation so reuse the one method
+        if field == 'members':
+            from . import resources
+
+            members = {
+                str(member.individual_guid): {'roles': member.roles}
+                for member in obj.members
+            }
+            for member_guid, roles in value.items():
+                members[member_guid] = roles
+
+            resources.validate_members(members)
```

Comment:
resources validate_member and validate_members were made separate functions to allow the usage below.
As that  has been removed, they can now be simplified.

Reply:
I merged `validate_member` into `validate_members` but that was about it :joy:

Reply:
Ah yes, you did. Ignore me :)

Reply:
No I just force pushed 🙂